src/cpp/fastdds/domain/DomainParticipantImpl.cpp

```diff
@@ -1447,7 +1447,7 @@ ReturnCode_t DomainParticipantImpl::register_type(
     std::lock_guard<std::mutex> lock(mtx_types_);
     types_.insert(std::make_pair(type_name, type));
 
-    if (type->auto_fill_type_object() || type->auto_fill_type_information())
+    if (type->auto_fill_type_information())
     {
         register_dynamic_type_to_factories(type);
     }
```

Comment:
This logic should be refactored (and probably removed). I suggest leaving a TODO(XTypes) to easily find it in the future.

Reply:
Commeted in [7dafd25](https://github.com/eProsima/Fast-DDS/pull/4126/commits/7dafd25b2d97a27f2f53acf522b0d2c68e72a46e)